src/OrchardCore.Modules/OrchardCore.AdminMenu/Models/AdminMenu.cs

```diff
@@ -9,8 +8,7 @@ public class AdminMenu
         public string Id { get; set; } = Guid.NewGuid().ToString("n");
         public string Name { get; set; }
         public bool Enabled { get; set; } = true;
-        public List<AdminNode> MenuItems { get; } = new List<AdminNode>();
-
+        public ImmutableArray<AdminNode> MenuItems { get; set; } = ImmutableArray.Create<AdminNode>();
```

Comment:
Why iimutable?

Reply:
In `AdminMenuService` we hold through a tenant **singleton memory cache an `AdminMenuList`** that holds a collection of `AdminMenu`, each of them holding a collection of `AdminNode`. **And they can be mutated**.

Then each `AdminNode` inherits from `MenuItem` that also has a child collection of `MenuItem`.

Because `MenuItem` is used in many place, to not change too much code, in place of modifying the `MenuItem` definition, e.g by using an immutable collection for its child items, i updated some `AdminNode` methods by doing a mutation on a new list of child items.